### PR TITLE
Use reselect everywhere

### PR DESCRIPTION
**Pull request by gshilin**
Also fix formatting problems

### CODE REVIEW

File: src/redux/createStore.js [hunk 1]
```diff
@@ -36,7 +36,7 @@ import player from './modules/player';
 import playlist from './modules/playlist';
 import fetchImage from './modules/fetchImage';
 
-const isProduction = process.env.NODE_ENV === 'production';
+const isProduction = true; // TODO gshilin process.env.NODE_ENV === 'production';
```
Comment (davgur) [inline src/redux/createStore.js:39]: is its ok?

File: src/redux/selectors.js [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+// Select additional parameters
+const _2 = (_, param) => param;
+const _3 = (_, __, param) => param;
+const _4 = (_, __, ___, param) => param;
+
+// auth
+export const authGetUserSelector = createSelector([getAuth], auth => authSel.getUser(auth));
+
+// settings
+export const settingsGetContentLanguagesSelector = createSelector(getSettings, _2, (s, skip) => settingsSel.getContentLanguages(s, skip));
```
Comment (bbfsdev) [inline src/redux/selectors.js:67]: The magic here is createSelector?
If the result is cached, when will it be invalidated?

Comment (bbfsdev) [reply src/redux/selectors.js:67]: I don't understand how this works.

```
const getContentLanguages = (state, skipFlags) => {
  if (state.urlLanguage.length && !skipFlags) {
    return state.urlLanguage;
  }

  if (state.showAllContent && !skipFlags) {
    return ALL_LANGUAGES;
  }

  return state.contentLanguages;
};
```

So what `createSelector` does? When will it know to invalidate the result and fetch getContentLanguages again?

**Review by davgur — APPROVED** [review]

File: src/redux/selectors.js (hunk 2 above)
Comment (gshilin) [reply src/redux/selectors.js:67]: read here https://reselect.js.org/introduction/how-does-reselect-work#why-reselect-is-often-used-with-redux

File: src/redux/createStore.js (hunk 1 above)
Comment (gshilin) [reply src/redux/createStore.js]: restored

**Pull request merged**